### DISCUSSION TITLE
Collecting the trace output

### DISCUSSION THREAD

**Discussion by murasu** [harfbuzz/harfbuzz#2865] [Q&A] [answered]
I like to work with the trace data we get from hb_shape command line utility. However, I need to get the data internally, with no access to the command line. How can I get this with just the APIs availabe in libharfbuzz.a ?

**Comment by behdad** [comment, marked as answer]
`hb_buffer_set_message_func()`.

**Reply by murasu** [reply]
Thanks for the precise starting point 👍
